[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/jinsanity07git/livedc/blob/main/notebooks/heartrate.ipynb#scrollTo=XgCQhLhv-Av0)

In [1]:
import os; import sys; os.system("pip install livedc") if "COLAB_JUPYTER_TRANSPORT" in os.environ else sys.path.insert(0, '../src')

In [2]:
from livedc.health.workout import fat_burning_heart_rate
fat_burning_heart_rate.rdoc()

The fat-burning zone is typically 50-70% of the maximum heart rate, which is calculated as:
    $$ \text{Maximum Heart Rate} = 220 - \text{Age} $$
    The lower and upper bounds of the fat-burning heart rate range are then calculated as:
    $$ \text{Lower Bound} = \text{Maximum Heart Rate} \times 0.5 $$
    $$ \text{Upper Bound} = \text{Maximum Heart Rate} \times 0.7 $$

    Args:
        age (int): Age of the person.
        
    Returns:
        tuple: A tuple containing the lower and upper bounds of the heart rate range.
    

In [3]:
age = 30
heart_rate_range = fat_burning_heart_rate(age)
mntime = 35
avgbpm  = 160
print(f"Fat-burning heart rate range for age {age}: {heart_rate_range}")

Fat-burning heart rate range for age 30: (95.0, 133.0)


In [4]:
prompt = f"If a {age}-year-old man jogs for {mntime} minutes with an average heart rate of {avgbpm} bpm," + \
            f"with a fat-burning zone of {heart_rate_range} bpm, what likely happens?"
print (prompt)

If a 30-year-old man jogs for 35 minutes with an average heart rate of 160 bpm,with a fat-burning zone of (95.0, 133.0) bpm, what likely happens?


LLM advice DEMO:

If a 30-year-old man jogs for 35 minutes with an **average heart rate of 160 bpm**, which is well **above his fat-burning zone of 95–133 bpm**, here's what likely happens:

### 🔥 1. **More Calories Burned, Less Fat Percentage**

- At **160 bpm**, he's exercising in the **cardio or high-intensity zone**, likely around **80–90% of his max heart rate** (220 - 30 = 190 bpm max).
- This zone **burns more total calories**, but the **percentage of fat used for fuel is lower** than in the fat-burning zone. The body relies more on **glycogen (carbs)** than fat at this intensity.

### ⚙️ 2. **Improved Cardiovascular Fitness**

- Sustaining that HR for 35 minutes builds **aerobic endurance** and improves **cardiovascular health**.
- It can also increase VO₂ max over time (your body’s ability to use oxygen efficiently).

### 💪 3. **Muscle Fatigue & Recovery Needs**

- At that intensity, muscles work harder, which leads to **higher lactate production**, possible **soreness**, and a **longer recovery period** compared to lower-intensity fat-burning exercise.

### ⚠️ 4. **Risk Consideration**

- If he’s **not used to high-intensity workouts** or has any **underlying health conditions**, sustaining 160 bpm for 35 minutes could cause:
  - Excessive strain on the heart
  - Dehydration or overheating
  - Overtraining symptoms if done too often

### ✅ Summary:

- **Pro**: Great for cardio endurance and total calorie burn.
- **Con**: Not the most efficient for fat-burning percentage-wise, and it's more demanding on the body.

Would you like help with planning a workout routine depending on your goal—like fat loss vs endurance?